README.md

```diff
@@ -1,2 +1,133 @@
-# notation-hashicorp-vault
-HashiCorp Vault provider for Notation CLI
+# notation-hc-vault
+HashiCorp Vault signing plugin for (Notation)[https://notaryproject.dev/]
+
+## Setup
+1. There are two binaries in the release: `key-helper` and `notation-hc-vault`.
+2. Install the `notation-hc-vault` plugin to the notation path specified by [notation plugin spec](https://github.com/notaryproject/notaryproject/blob/main/specs/plugin-extensibility.md#installation). On unix, the path is `$XDG_CONFIG_HOME/notation/plugins/hc-vault/notation-hc-vault`.
+3. `key-helper` is a standalone helper CLI used to configure Hashicorp Vault.
+
+## Install and Configure Hashicorp Vault Server
+1. [Install Hashicorp Vault](https://developer.hashicorp.com/vault/downloads)
+2. Create a vault config file under your work directory. A minimal working example is shown below, more details about config can be found [here](https://developer.hashicorp.com/vault/tutorials/operations/configure-vault#configuration-files)):
+    ```bash
+    $ cat > vault-server.hcl <<EOF
+    disable_mlock = true
+    ui            = true
+
+    listener "tcp" {
+    address     = "127.0.0.1:8200"
+    tls_disable = "true"
+    }
+
+    storage "file" {
+    path = "/tmp/vault-data"
+    }
+    EOF
+    ```
+3. Start a prod server 
+    ```bash 
+    vault server -config vault-server.hcl
+    ```
+   Note: when this process is stopped, the vault will be sealed. To unseal the vault, you need to run this command again.
+4. At this point, open `127.0.0.1:8200` in your browser to see the webpage of your Hashicorp Vault server.
+5. Initialize the vault (this is a one-time step, you do not need to do this step in the future). In this example, the root key is split into 3 key shares, and any two keys of the three will be sufficient to unseal the Vault, [see details](https://developer.hashicorp.com/vault/docs/concepts/seal).
+    ![](./docs/root_keys.png)
+    ![](./docs/download_keys.png)
+    
+    ! IMPORTANT !: Remember to hit the `Download Keys` button before going forward. Both keys and the initial root token are downloaded. They are required to unseal the vault and sign in.
+    
+    Unseal the Vault <br>
+    Enter two of the three keys generated previously to unseal the vault.
+    ![](./docs/unseal_vault.png)
+    
+    Sign in <br>
+    For test purpose, enter the root token in `Token`
+    ![](./docs/sign_in.png)
+    
+    By now, the Hashicorp Vault production server is ready to be used.
+6. Configure the Vault client to talk to your server:
+    ```bash
+    export VAULT_ADDR='http://127.0.0.1:8200'
+    ```
+    Set the `VAULT_TOKEN` environment variable value to the downloaded Root Token value.
+    ```bash
+    export VAULT_TOKEN="hvs.**************"
+    ```
```

Comment:
```suggestion
    ```
    Note: Do not use the root token for production usage of Notary and Vault.
```

Reply:
Added to the doc.